mesos-client/src/main/scala/com/mesosphere/mesos/client/CredentialsProvider.scala

```diff
@@ -0,0 +1,78 @@
+package com.mesosphere.mesos.client
+
+import java.net.URL
+import java.time.Instant
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{BasicHttpCredentials, GenericHttpCredentials, HttpCredentials}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
+import org.json4s.native.JsonMethods
+import pdi.jwt.JwtJson4s
+import pdi.jwt.JwtAlgorithm.RS256
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+trait CredentialsProvider {
+
+  def tokens(): Source[HttpCredentials, NotUsed]
+}
+
+case class JwtProvider(uid: String, privateKey: String, root: URL)(implicit system: ActorSystem, materializer: ActorMaterializer, context: ExecutionContext) extends CredentialsProvider {
+  import org.json4s._
+  import org.json4s.JsonDSL._
+  import JsonMethods.{parse, render, compact}
+
+  private def expireIn(duration: Duration): Long = Instant.now.getEpochSecond + duration.toSeconds
+  private val claim = JObject(("uid", uid), ("exp", expireIn(60.minutes))) // TODO: configure token expiration.
```

Comment:
I really like the stabby-arrow syntax for tuples in this context:

```suggestion
  private val claim = JObject("uid" -> uid, "exp" -> expireIn(60.minutes)) // TODO: configure token expiration.
```